README.md

```diff
@@ -1,3 +1,337 @@
+
+Shows the signature of the person who signed the version npecl would install, or from the specific version, if given.
+
+#### `upgrade {?{ext-name}{?:version-constraint}{?@dev-branch-name}}`
+
+Attempts to upgrade all installed versions to the latest available ones on GitHub (unless their major version has
+changed) If a specific package is specified, only attempt to upgrade that specific one.
+
+### CLI Options
+
+#### `--dry-run`
+
+Shows what it is about to do, but doesn't actually install
+
+#### `--force`
+
+To attempt to install a version that doesn't match the version constraints from the meta-data, for instance to install
+an older version than recommended, or when the signature is not available.
+
+#### `--pre-release`
+To allow for installation of pre-release versions
+
+#### `--with-php-config=/path/to/php-config`
+
+Allows installation of extensions with PHP versions that are not in the path
+
+#### `--{option}{?=value}`
+
+All options specified in the `config` section of the composer.json file can also be given, including a value if they
+take them. For example, for Xdebug you could run `npecl install xdebug --without-xdebug-compression`.
+
+## Extension maintainer: register an npecl package
+
+Create a `composer.json` in your repository, commit:
+
+```json
+{
+    "name": "ext-xdebug",
+    "type": "php-ext-zend",
+    "license": "Xdebug-1.03",
+    "description": "Xdebug is an amazing tool...",
+    "require": {
+        "ext-zlib": "*",
+        "php": ">=7.4.21,<8.4"
+    },
+    "conflict": {
+        "ext-zetalib": "*"
+    },
+    "php-ext": {
+        "priority": 90,
+        "config": {
+            "enable-xdebug-dev": false,
+            "without-xdebug-compression": false
+        }
+    }
+}
+```
+
+Submit your repo URL into [https://packagist.org/packages/submit](https://packagist.org/packages/submit). Packagist
+will read the `composer.json` as usual.
+
+When releases are made, Packagist will do the usual thing of producing metadata for the release
+
+### Windows binaries
+
+Windows needs pre-built binary DLLs. The expected workflow is that the release is made, then some kind of build takes
+place (for example, in a GitHub Action, or manually on a compatible build environment), and the `.dll` file is added to
+the GitHub release.
+
+The name for the DLLs must follow the following pattern:
+
+ * `php_{extension-name}-{tag}-{php-maj/min}-{compiler}{"-nts"}?{-arch}?.dll`
+
+The descriptions of these items:
+
+ * `extension-name` the name of the extension, e.g. `xdebug` - this is the same as defined in `composer.json` minus `ext-`
+ * `tag` for example `3.3.0alpha3` - defined by the tag/release you have made
+ * `php-maj/min` - for example `8.3` for PHP 8.3.*
+ * `compiler` - usually `vc16`?
+ * `-nts` - optional - non-thread safe. If omitted, thread-safe mode is assumed
+ * `-arch` - optional - for example `x86_64`, fetch using `uname -i`. If omitted, the extension is assumed to be
+   compatible with all architectures
```

Comment:
I think maybe the `arch` should be required then; if people want to offer/download 32-bit builds specifically, that should be explicit, rather than assumed "no arch is 32-bit"; as generally I'd assume "no arch means it works on all architectures" :grin: arm64 is gaining popularity now too, so we'd have to figure out what the various archs report on OSX/Linux/Unix/Windows etc. :roll_eyes: Surely someone has already done this research? I'll have a look.

Reply:
Yeah, I would just have:

- `x86_64`
- `x86_32` (or `x86`)
- `arm64`

FWIW, on my home machine, `uname -i` shows `unknown`. `uname -m` shows `x86_64`.